tests/src/Examples.hs

```diff
@@ -0,0 +1,109 @@
+
+-- | The examples for the `Backends` module.
+solverTests :: TestTree
+solverTests =
+  testGroup
+    "API use examples"
+    [ testCase "basic use" solverBasicUse
+    ]
+
+-- | Basic use of the `Solver` datatype.
+solverBasicUse :: IO ()
+solverBasicUse =
+  -- we need a solver backend to evaluate our commands
+  -- examples on how to create such backends can be found in the corresponding
+  -- sections, e.g. `processBasicUse`
+  withBackend $ \backend -> do
+    -- then, we create a solver out of the backend
+    -- we enable queuing (it's faster !)
+    solver <- initSolver backend True
+    -- we send a basic command to the solver and ignore the response
+    -- we can write the command as a simple string because we have enabled the
+    -- OverloadedStrings pragma
+    _ <- command solver "(get-info :name)"
+    return ()
+  where
+    withBackend :: (Backend -> IO a) -> IO a
+    withBackend = Process.with def . (. Process.toBackend)
+
+-- | The examples for the `Process` backend.
+processTests :: TestTree
+processTests =
+  testGroup
+    "API use examples"
+    [ testCase "basic use" processBasicUse,
+      testCase "setting options" processSetOptions,
+      testCase "exiting manually" processManualExit
+    ]
+
+-- | Basic use of the `Process` backend.
+processBasicUse :: IO ()
+processBasicUse =
+  -- `Process.with` runs a computation using the `Process` backend
+  Process.with
+    -- the configuration type `Process.Config` is an instance of the `Default` class
+    -- we can thus use a default configuration for the backend with the keyword `def`
+    -- this default configuration uses Z3 as an external process and disables logging
+    def
+    $ \handle -> do
+      -- first, we make the process handle into an actual backend
+      let backend = Process.toBackend handle
+      -- then, we create a solver out of the backend
+      -- we enable queuing (it's faster !)
+      solver <- initSolver backend True
+      -- we send a basic command to the solver and ignore the response
+      -- we can write the command as a simple string because we have enabled the
+      -- OverloadedStrings pragma
+      _ <- command solver "(get-info :name)"
+      return ()
+
+-- | An example of how to set options for the `Process` backend.
+processSetOptions :: IO ()
+processSetOptions =
+  -- here we use a custom-made configuration
+  let myConfig =
+        Process.Config
+          { Process.exe = "z3",
+            Process.args = ["-in", "solver.timeout=10000"],
+            Process.reportError = LBS.putStr . (`LBS.snoc` '\n')
+          }
+   in Process.with myConfig $ \handle -> do
+        -- since the `Process` module exposes its `Handle` datatype entirely, we can also
+        -- set the options of the underlying process
+        let p = Process.process handle
+            stdin = getStdin p
+        -- for instance here we change the buffering mode of the process' input channel
+        hSetBuffering stdin LineBuffering
```

Comment:
There is at least one use case: liquid-fixpoint sets the buffering mode to `BlockBuffering $ Just $ someNumber` whereas we default it to `BlockBuffering Nothing`. I didn't add this specific example because it was taking more space, but I can add a line explaining we already have good defaults for these options and they probably don't need to be changed.